(feat): Add gasket-plugin-happyfeet (#595)
godaddy/gasket: packages/gasket-plugin-happyfeet/README.md

Reviewer: I'd recommend adding a usage example that shows how you can call `gasket.happyFeet` directly to manually change the health status (this allows custom termination logic for things not included in the default metrics monitoring).

# @gasket/plugin-happy-feet

> A plugin that creates a [happy-feet](https://github.com/asilvas/happy-feet#usage) instance and adds pod memory healthchecks based on config that you provide.

Reviewer: Should be generalized since it does more than monitor memory, and it works for more than just Kubernetes.

```suggestion
Integrates with [happy-feet](https://github.com/asilvas/happy-feet#usage) to add application instance termination when health metrics such as memory usage enter a bad state.
```

## Installation

#### New apps

```

> gasket create <app-name> --plugins @gasket/plugin-happy-feet

Reviewer 2: Need the naming in the example to be consistent with the package.

```suggestion
gasket create <app-name> --plugins @gasket/plugin-happyfeet
```

```

#### Existing apps

```
npm i @gasket/plugin-happy-feet
```

> Modify `plugins` section of your `gasket.config.js`:

Reviewer: ```suggestion
Modify the `plugins` section of your `gasket.config.js`:
```

```diff
module.exports = {
  plugins: {
    add: [

> +      '@gasket/plugin-happy-feet'

Reviewer 2: ```suggestion
+      '@gasket/plugin-happyfeet'
```

    ]
  }
}
```

## Configuration

You can specify the various [happy-feet](https://github.com/asilvas/happy-feet#usage) config options in `gasket.config.js`.

```js
// gasket.config.js
module.exports = {
   happyFeet: {
    // https://github.com/asilvas/happy-feet#usage
    escalationSoftLimitMin: 20, // 20s
    escalationSoftLimitMax: 300, // 5min
    uncaughtExceptionSoftLimit: 1,
    uncaughtExceptionHardLimit: void 0,
    rssSoftLimit: 0.9 * 1024 * 1024 * 1024, // ~900MB
    rssHardLimit: 1.8 * 1024 * 1024 * 1024, // ~1.9GB
    logOnUnhappy: true
  }
};
```

> [Happy Feet] for gasket has no default configuration.

Reviewer: Link is missing a URL.

# @gasket/plugin-happyfeet

Integrates with [happy-feet](https://github.com/asilvas/happy-feet#usage) to add application instance termination when health metrics such as memory usage enter a bad state.

## Installation

#### New apps

```
gasket create <app-name> --plugins @gasket/plugin-happyfeet
```

#### Existing apps

```
npm i @gasket/plugin-happyfeet
```

Modify the `plugins` section of your `gasket.config.js`:

```diff
module.exports = {
  plugins: {
    add: [
+      '@gasket/plugin-happyfeet'
    ]
  }
}
```

## Configuration

You can specify the various [happy-feet](https://github.com/asilvas/happy-feet#usage) config options in `gasket.config.js`.

```js
// gasket.config.js
module.exports = {
   happyFeet: {
    // https://github.com/asilvas/happy-feet#usage
    escalationSoftLimitMin: 20, // 20s
    escalationSoftLimitMax: 300, // 5min
    uncaughtExceptionSoftLimit: 1,
    uncaughtExceptionHardLimit: void 0,
    rssSoftLimit: 0.9 * 1024 * 1024 * 1024, // ~900MB
    rssHardLimit: 1.8 * 1024 * 1024 * 1024, // ~1.9GB
    logOnUnhappy: true
  }
};
```

[Happy Feet](https://github.com/asilvas/happy-feet#usage) for gasket has no default configuration.

> ### Usage
> You can call gasket.happyFeet directly to manually change the health status for any reason by 
> setting `gasket.happyFeet.status = 'unhappy'` in any context that you have the gasket object available to your function. 
> For example, you can define middleware to run before any healthchecks to define custom termination logic for things not included in the default metrics monitoring.
> ## License
> [MIT](./LICENSE.md)

Reviewer 2: nit: tweak heading level and add some whitelines

```suggestion
## Usage

You can call gasket.happyFeet directly to manually change the health status for any reason
by setting `gasket.happyFeet.status = 'unhappy'` in any context that you have the gasket
object available to your function. 

For example, you can define middleware to run before any healthchecks to define custom
termination logic for things not included in the default metrics monitoring.

## License

[MIT](./LICENSE.md)
```
